### ISSUE TITLE
Connection don't always get rendered.

### ISSUE THREAD

**Issue by navanshu** [chrisjpatty/flume#117]
On react v17.0.2 all connection don't get rendered. The nodes that are close to the start shows connection but the nodes far from center of the board don't get rendered.

**Comment by Dante83** [comment]
I am also having this issue when I switch between flume graphs. I used a key to trigger re-renders, as described in 

https://github.com/chrisjpatty/flume/issues/71

which works great, but all of my connections are lost and when I go to add connections, the connections show as hooking into random positions in space.

**Comment by chrisjpatty** [comment]
Hey thanks for reporting, do you have a repo with this problem that I could use to debug it? Thanks!

**Comment by navanshu** [comment]
I have like 300 nodes per graph and as you go to the right from center point of the artboard connections disappear

**Comment by Dante83** [comment]
Ok! I went through and managed to create a new example project and sliced down a lot of the fluff that's in the primary project I'm working on. Still not sure if this might be a different problem then what navanshu is speaking of, but maybe they're related.

https://github.com/Dante83/broken-flume-connections

I've made an example of what I've been doing there. The basics of my strategy should be straight from the examples I've seen.

1.  Grab the current node state with `nodeEditor.current.getNodes()`.
2.  Save this json object to the redux slice object for future use.
3.  Grab this slice when we need it and set the value of the nodes attribute in `<NodeEditor/>`.

I do something similiar for comments. It does seem like the information is saving, but the connections aren't rendering and once it breaks, it breaks all future connections.

Fingers crossed that this will help, really awesome project, I'm looking forward to putting some logic behind these little nodes.

**Comment by navanshu** [comment]
@Dante83 @chrisjpatty 
Hi, I think the error is originating from d3. 
The connections are there it's just that at some zoom levels they don't show up.
I have to zoom out pretty far to see the connections. 
The connections at the start of the graph won't have any problems but the ones towards the end have issues.

**Comment by Dante83** [comment]
@navanshu Yep, my issues seems to be a bit different, maybe I'm doing something wrong though. The lines aren't appearing at any zoom level, really. But when connect new ones to the reset state after loading the nodes, I find that there is a 'line' but it's really just a little white dot in the far left corner that moves further up and left as I move the nodes around. I suspect it's setting the starting and ending locations wrong when I reload these and that's what's driving my issue.
